querybook/server/lib/result_store/stores/file_store.py

```diff
@@ -27,25 +27,38 @@ def __init__(self, uri: str):
         self.uri = get_file_uri(uri)
 
     def start(self):
+        self._chunk_size = QuerybookSettings.FILE_CHUNK_SIZE
         self._chunks_length = 0
+        self._chunks = []
         os.makedirs(self.uri_dir_path, exist_ok=True)
 
     def write(self, data: str):
-        # write each line into csv
+        # Append each line to a list and then dump at once in batches
         data_len = len(data)
         if (
             QuerybookSettings.DB_MAX_UPLOAD_SIZE > 0
             and self._chunks_length + data_len > QuerybookSettings.DB_MAX_UPLOAD_SIZE
         ):
             return False
-
+        
         self._chunks_length += data_len
+        
+        self._chunks.append(data)
+
+        if len(self._chunks) == self._chunk_size:
```

Comment:
Just realized that the `DB_MAX_UPLOAD_SIZE` is the actual data size, while this `_chunk_size` is the number of records. we may want to make the chunk size based on the data size as well (multiples of 1024)

Reply:
Hi @jczhong84,
Thanks for reviewing my fix. 

A question that popped in my head is , below check remains valid for dbstore as we are making entry to database. In filestore, as we are writing to file , Do we need the same? 
if (
            QuerybookSettings.DB_MAX_UPLOAD_SIZE > 0
            and self._chunks_length + data_len > QuerybookSettings.DB_MAX_UPLOAD_SIZE
        ):
            return False

Reply:
thanks for pointing this out. 
could you add a separate max size setting for the file store? 
can set a big number for it as default, or just set 0 as unlimited by default.

Reply:
hi @jczhong84,
Defining the max size of file to 0 does not work because even if the number of rows returned from the query is 0, it will anyhow write the column names to the file. 

But if we are setting it to a high value,  Im wondering does it serve the purpose of choosing filestore above dbstore? Correct me if im wrong.

Reply:
if the query result has 0 rows, isn't it expected to write a empty file with only the column names?
e.g. set DB_MAX_UPLOAD_SIZE as 0, it will bypass below check, which basically means no limit check.
```
        if (
            QuerybookSettings.DB_MAX_UPLOAD_SIZE > 0
            and self._chunks_length + data_len > QuerybookSettings.DB_MAX_UPLOAD_SIZE
        ):
            return False
```

what matters is the file size, not the number of rows. what I meant is  by default set it as 0 (unlimited), but people can set their own limit like 10GB to override this default setting

Reply:
Hi @jczhong84,
Have made the amendments as suggested.

Regards,
Najma

Reply:
thanks for the update.

use 
`if _chunks_length >= self._chunk_size:`
but not
`if len(self._chunks) == self._chunk_size:`

Reply:
hi @jczhong84 ,
 have changed the code and logic to use self._chunks_length.